app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListViewModel.kt

```diff
@@ -0,0 +1,213 @@
+import com.saltedge.authenticator.sdk.model.connection.ConnectionStatus
+import com.saltedge.authenticator.sdk.model.connection.getStatus
+import com.saltedge.authenticator.sdk.model.connection.isActive
+import com.saltedge.authenticator.sdk.model.error.ApiErrorData
+import com.saltedge.authenticator.sdk.tools.keystore.KeyStoreManagerAbs
+import javax.inject.Inject
+
+class ConnectionsListViewModel @Inject constructor(
+    private val appContext: Context,
+    private val connectionsRepository: ConnectionsRepositoryAbs,
+    private val keyStoreManager: KeyStoreManagerAbs,
+    private val apiManager: AuthenticatorApiManagerAbs
+) : ViewModel(), LifecycleObserver, ConnectionsRevokeListener {
+
+    var onQrScanClickEvent = MutableLiveData<ViewModelEvent<Unit>>()
+        private set
+    var onListItemClickEvent = MutableLiveData<ViewModelEvent<Int>>()
+        private set
+    var onSupportClickEvent = MutableLiveData<ViewModelEvent<String?>>()
+        private set
+    var onReconnectClickEvent = MutableLiveData<ViewModelEvent<String>>()
+        private set
+    var onRenameClickEvent = MutableLiveData<ViewModelEvent<Bundle>>()
+        private set
+    var onDeleteClickEvent = MutableLiveData<ViewModelEvent<Bundle>>()
+        private set
+
+    val listVisibility = MutableLiveData<Int>()
+    val emptyViewVisibility = MutableLiveData<Int>()
+    val reconnectViewVisibility = MutableLiveData<Int>()
+    val deleteTextViewResource = MutableLiveData<Int>()
+    val deleteImageViewResource = MutableLiveData<Int>()
+
+    val listItems = MutableLiveData<List<ConnectionViewModel>>()
+    val listItemsValues: List<ConnectionViewModel>
+        get() = listItems.value ?: emptyList()
+    val updateListItemEvent = MutableLiveData<ViewModelEvent<Int>>()
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_START)
+    fun onStart() {
+        updateViewsContent()
+    }
+
+    override fun onConnectionsRevokeResult(revokedTokens: List<Token>, apiError: ApiErrorData?) {}
+
+    fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (data == null || resultCode != Activity.RESULT_OK) return
+        val listItem = listItemsValues.find { it.guid == data.getStringExtra(KEY_GUID) } ?: return
+        when (requestCode) {
+            RENAME_REQUEST_CODE -> {
+                onUserRenamedConnection(
+                    listItem = listItem,
+                    newConnectionName = data.getStringExtra(KEY_NAME) ?: return
+                )
+            }
+            DELETE_REQUEST_CODE -> onUserConfirmedDeleteConnection(listItem = listItem)
+        }
+    }
+
+    fun onViewClick(viewId: Int) {
+        if (viewId == R.id.actionView) onQrScanClickEvent.postValue(ViewModelEvent(Unit))
+    }
+
+    fun onListItemClick(itemIndex: Int) {
+        onListItemClickEvent.postValue(ViewModelEvent(itemIndex))
+    }
+
+    fun checkStatusOfConnection(itemIndex: Int) {
+        if (isActiveConnection(itemIndex)) {
```

Comment:
maybe better to add this fields as fields of ConnectionViewModel. 
e.g. `hasReconnect`,  `deleteMenuItemText`, `deleteMenuItemImage`